docs/index.md

```diff
@@ -30,92 +28,160 @@ This means that you can write your entire application in a single language, Rust
+
+### Table
+A SpacetimeDB **table** is a database table. Tables are declared in a module's native language. For instance, in Rust, a table is declared like so:
+
+```csharp
+[SpacetimeDB.Table(Name = "players", Public = true)]
+public partial struct Player
+{
+    [SpacetimeDB.PrimaryKey]
+    uint playerId;
+    string name;
+    uint age;
+    Identity user;
+}
+```
+<!-- TODO: switchable language widget.
+```rust
+#[spacetimedb::table(name = players, public)]
+pub struct Player {
+   #[primary_key]
+   id: u64,
+   name: String,
+   age: u32,
+   user: Identity,
+}
+```
+-->
+
+The contents of a table can be read and updated by [reducers](#reducer).
+Tables marked `public` can also be read by [clients](#client).
+
+### Reducer
+A **reducer** is a function exported by a [module](#module).
+Connected [clients](#client-side-sdks) can call reducers to interact with the module.
+This is a form of [remote procedure call](https://en.wikipedia.org/wiki/Remote_procedure_call).
+Reducers can be invoked across languages. For example, a Rust [module](#module) can export a reducer like so:
+
+```csharp
+[SpacetimeDB.Reducer]
+public static void SetPlayerName(ReducerContext ctx, uint playerId, string name)
+{
+    // ...
+}
+```
+<!-- TODO: switchable language widget.
+```rust
+#[spacetimedb::reducer]
+pub fn set_player_name(ctx: &spacetimedb::ReducerContext, id: u64, name: String) -> Result<(), String> {
+   // ...
+}
+```
+-->
+
+And a C# [client](#client) can call that reducer:
+
+```cs
+void Main() {
+   // ...setup code, then...
+   Connection.Reducer.SetPlayerName(57, "Marceline");
+}
+```
+
+These look mostly like regular function calls, but under the hood, the client sends a request over the internet, which the module processes and responds to.
+
+The `ReducerContext` passed into a reducer includes information about the caller's [identity](#identity) and [address](#address).
+It also allows accessing the database and scheduling future operations.
+
+### Client
+A **client** is an application that connects to a [module](#module). A client logs in using an [identity](#identity) and receives an [address](#address) to identify the connection. After that, it can call [reducers](#reducer) and query public [tables](#table).
+
+Clients are written using the [client-side SDKs](#client-side-sdks). These are regular software libraries. Unlike the [server-side libraries](#server-side-libraries), they don't require the use of any special build tools.
+
+### Identity
+
+A SpacetimeDB `Identity` identifies someone interacting with a module. It is a long lived, public, globally valid identifier that will always refer to the same end user, even across different connections.
+
+If the end user authenticates through an [OpenID Connect](https://openid.net/developers/how-connect-works/) provider, such as Google or Facebook, their `Identity` will be tied to the account they use on that provider.
```

Comment:
@gefjon please review this verbiage

Reply:
I like this. Using the OIDC words is accurate and sounds good.